internal/service/configuration_reader.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// ArrayConnectionData contains data required to connect to array
+type ArrayConnectionData struct {
+	SystemID  string `json:"systemID"`
+	Username  string `json:"username"`
+	Password  string `json:"password"`
+	Endpoint  string `json:"endpoint"`
+	Insecure  bool   `json:"insecure,omitempty"`
+	IsDefault bool   `json:"isDefault,omitempty"`
+}
+
+// ConfigurationReader handles reading of the storage system configuration secret
+type ConfigurationReader struct{}
+
+// GetStorageSystemConfiguration returns a storage system from the configuration file
+// If no default system is supplied, the first system in the list is returned
+func (c *ConfigurationReader) GetStorageSystemConfiguration(file string) (*ArrayConnectionData, error) {
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		return nil, fmt.Errorf(fmt.Sprintf("File %s does not exist", file))
+	}
+
+	config, err := ioutil.ReadFile(filepath.Clean(file))
+	if err != nil {
+		return nil, fmt.Errorf(fmt.Sprintf("File %s errors: %v", file, err))
+	}
+
+	var storageSystem ArrayConnectionData
+
+	if string(config) == "" {
+		return nil, fmt.Errorf("arrays details are not provided in vxflexos-config secret")
```

Comment:
The first letter case in the error messages is inconsistent. Shouldn't all messages start from a lowercase letter?